### ISSUE TITLE
Add an option to control the mails cache size

### ISSUE THREAD

**Issue by DenBond7** [FlowCrypt/flowcrypt-android#669]
#668 #585

**Comment by DenBond7** [comment]
@tomholub I need to know your opinion. Currently, the app uses a cache which uses [DiskLruCache](https://github.com/JakeWharton/DiskLruCache). The cache size = 50Mb. I think for the first time that's enough. As for me, it'll be better to move that to the next releases. In this issue, I was going to add UI, where a user can change the cache size. But maybe it's redundant for the current realization. Maybe 50Mb it's enough for most cases.

**Comment by tomholub** [comment]
I was going to ask you about this. How is the cache saved - does Android system know what this is deletable cache? If I go to android settings -> flowcrypt -> delete cache, will the cache get wiped?

**Comment by tomholub** [comment]
Also, do the oldest unused cache records automatically get replaced with the new ones to stay under 50mb?

**Comment by DenBond7** [comment]
@tomholub 
> How is the cache saved - does Android system know what this is deletable cache? If I go to android settings -> flowcrypt -> delete cache, will the cache get wiped?

No, the cache won't wiped. I use the [private app directory](https://developer.android.com/training/data-storage/files/internal) to store the cache.  I use `getFilesDir()` to get a directory. Then this cache can be controlled only by app.

> Also, do the oldest unused cache records automatically get replaced with the new ones to stay under 50mb?

Of course, it's how LruDiskCache works

**Comment by DenBond7** [comment]
Least Recently Used (LRU)

**Comment by tomholub** [comment]
Is there no way we can store the cache in a place that Android can understand and delete as needed, while still using LruDiskCache?
